src/base/static/components/input-form/index.js

```diff
@@ -106,17 +102,21 @@ class InputForm extends Component {
   }
 
   setStageLayers(stageConfig) {
-    stageConfig.visible_layer_ids &&
-      this.props.showLayers(stageConfig.visible_layer_ids);
-    stageConfig.visible_layer_ids &&
-      this.props.hideLayers(
-        this.props.mapLayers
-          .filter(layer => !layer.is_basemap)
-          .filter(layer => !stageConfig.visible_layer_ids.includes(layer.id))
-          .map(layer => layer.id),
+    if (stageConfig.visibleLayerGroupIds) {
+      // Set layers for this stage.
+      stageConfig.visibleLayerGroupIds.forEach(layerGroupId =>
+        this.props.updateLayerGroupVisibility(layerGroupId, true),
       );
-    stageConfig.visible_basemap_id &&
-      this.props.setBasemap(stageConfig.visible_basemap_id);
+      // Hide all other layers.
+      this.props.mapConfig.layerGroups
+        .filter(
+          layerGroup =>
+            !stageConfig.visibleLayerGroupIds.includes(layerGroup.id),
+        )
+        .forEach(layerGroup =>
+          this.props.updateLayerGroupVisibility(layerGroup.id, false),
```

Comment:
I notice that we are using this pattern of firing the `updateLayerGroupVisbility` action creators within a forloop throughout several places in the app.

I think it would be simpler to have this action creator able to set the visibilities on multiple layer at once. For example, it can accept an array of object with the schema `{ layerGroupId: number, isVisible: bool }`. It would give us a more concise way of setting visibilities on multiple layers, while reducing the number of actions dispatched.

Not a big deal - but interested to know what your thoughts are.

Reply:
I think that makes sense. I'm wondering if it will be easier to refactor this though once we iron out the `layerGroups`/`layerGroupsMetadata` issue discussed elsewhere in this diff.

I think I'll collect those two TODOs under an issue and tackle them at a later step unless you think this should be addressed now.

Reply:
Sounds good :+1: